hy/compiler.py

```diff
@@ -1221,17 +1221,31 @@ def _compile_import(expr, module, names=None, importer=asty.Import):
     def compile_index_expression(self, expr):
         expr.pop(0)  # index
 
-        indices, ret, _ = self._compile_collect(expr[1:])
-        ret += self.compile(expr[0])
+        if isinstance(expr[1], HyExpression) and expr[1][0] == "slice":
```

Comment:
This will trigger on the string `"slice"` instead of just the symbol.  You may also need to check that `expr[1]` is nonempty.